### PR TITLE
add runtime validation in keadm

### PR DESCRIPTION
**Pull request by Shelley-BaoYue**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**

Add runtime validation in keadm. Return a more explicit error message.

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/edge/join.go [hunk 1]
```diff
@@ -253,6 +253,9 @@ func createEdgeConfigFiles(opt *common.JoinOptions) error {
 	if v.Major <= 1 && v.Minor < 12 {
 		return createV1alpha1EdgeConfigFiles(opt)
 	}
+	if v.Major >= 1 && v.Minor >= 14 && opt.RuntimeType != constants.DefaultRuntimeType {
+		return fmt.Errorf("since KubeEdge v1.14, runtime type only supports `remote`")
```
Comment (fisherxu) [inline keadm/cmd/keadm/app/cmd/edge/join.go:257]: If users want to use Docker, the runtime type still set to `remote`, then edged connect to `cri-dockerd`?

Comment (Shelley-BaoYue) [reply keadm/cmd/keadm/app/cmd/edge/join.go:257]: > If users want to use Docker, the runtime type still set to `remote`, then edged connect to `cri-dockerd`?

set runtimeType to remote and set `remoteRuntimeEndpoint` to `var/run/cri-dockerd.sock`, edged will connect to `cri-dockerd`.  See more detail in #4843

Comment (fisherxu) [reply keadm/cmd/keadm/app/cmd/edge/join.go:257]: Okey :)

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**